api_app/db/repositories/resources.py

```diff
@@ -124,6 +124,22 @@ async def patch_resource(self, resource: Resource, resource_patch: ResourcePatch
         await self.update_item_with_etag(resource, etag)
         return resource, resource_template
 
+    async def get_resource_dependecny_graph(self, resource: Resource) -> List:
+        # Get the parent resource path and id
+        parent_resource_path = resource.resourcePath
+        dependentResourcesList = []
+
+        # Get all related resources
+        related_resources_query = f"SELECT * FROM c WHERE CONTAINS(c.resourcePath, '{parent_resource_path}') AND c.deploymentStatus != '{Status.Deleted}' and c.deploymentStatus != '{Status.DeploymentFailed}'"
```

Comment:
should you sanitize the input before creating the query string?

Reply:
The input is the resource which is satanize by the method signature (line 127)